yarn-project/p2p/src/services/reqresp/connection-sampler/connection_sampler.ts

```diff
@@ -232,34 +234,37 @@ export class ConnectionSampler {
 
   /**
    * Closes a stream and updates the active connections count
-   *
-   * @param streamId - The stream id
    */
-  async close(streamId: string): Promise<void> {
+  async close(stream: Stream): Promise<void> {
+    let peerId = undefined;
+
     try {
-      const streamAndPeerId = this.streams.get(streamId);
-      if (!streamAndPeerId || !streamAndPeerId.stream) {
-        this.logger.debug(`Stream ${streamId} not found`);
-        return;
+      peerId = stream.metadata.peerId;
+      let updatedActiveConnectionsCount = undefined;
+
+      if (!peerId) {
+        this.logger.warn(`Stream ${stream.id} does not have a peerId set`);
+      } else {
+        updatedActiveConnectionsCount = (this.activeConnectionsCount.get(peerId) ?? 1) - 1;
+        this.activeConnectionsCount.set(peerId, updatedActiveConnectionsCount);
       }
 
-      const { stream, peerId } = streamAndPeerId;
-
-      const updatedActiveConnectionsCount = (this.activeConnectionsCount.get(peerId) ?? 1) - 1;
-      this.activeConnectionsCount.set(peerId, updatedActiveConnectionsCount);
-
       this.logger.trace('Closing connection', {
-        streamId,
+        streamId: stream.id,
         peerId: peerId.toString(),
         protocol: stream.protocol,
         activeConnectionsCount: updatedActiveConnectionsCount,
       });
 
-      await stream!.close();
+      if (!this.streams.has(stream)) {
+        this.logger.warn(`Stream ${stream.id} is not in the active streams set`);
```

Comment:
if the stream has been closed on the other end this will fire, which can happen in normal operation, i would drop this to debug